hail/src/test/scala/is/hail/expr/ir/TableIRSuite.scala

```diff
@@ -430,15 +430,34 @@ class TableIRSuite extends HailSuite {
   )
 
   @DataProvider(name = "join")
-  def joinData(): Array[Array[Any]] =
+  def joinData(): Array[Array[Any]] = {
+    val defaultLParts = 2
+    val defaultRParts = 2
+    val (defaultJ, defaultP) = joinTypes(0)
```

Comment:
Is this saying to only ever test outer joins?

Reply:
This change tests the full span of values within each parameter class (join types, for instance, but doesn't test against the full cartesian product.

You're right that it's probably worth doing the product over all join types, though. That should still be ~10x faster than the test was before.